frontend/src/components/activePage/TableInstance/RowInstanceActions/RowInstanceActionsPersistent.tsx

```diff
@@ -7,57 +7,97 @@ import {
+import { FetchResult } from '@apollo/client';
+import { ApplyInstanceMutation } from '../../../../generated-types';
 
 export interface IRowInstanceActionsPersistentProps {
   extended: boolean;
   instance: Instance;
-  startInstance: (idInstance: string, idTemplate: string) => void;
-  stopInstance: (idInstance: string, idTemplate: string) => void;
+  switchPersistent: (
+    instanceId: string,
+    tenantNamespace: string,
+    patchJson: string
+  ) => Promise<
+    FetchResult<ApplyInstanceMutation, Record<string, any>, Record<string, any>>
+  >;
 }
 
 const RowInstanceActionsPersistent: FC<IRowInstanceActionsPersistentProps> = ({
   ...props
 }) => {
-  const { extended, instance, startInstance, stopInstance } = props;
+  const { extended, instance, switchPersistent } = props;
 
-  const { status, idTemplate, name } = instance;
+  const { status, name, running, tenantNamespace } = instance;
 
   const font22px = { fontSize: '22px' };
 
+  const [disabled, setDisabled] = useState(false);
+
   return status === 'VmiReady' ? (
     <Tooltip placement="top" title={'Pause'}>
       <Button
+        loading={disabled}
         className={`hidden ${
           extended ? 'sm:block' : 'xs:block'
         } flex items-center`}
         type="warning"
         with="link"
         shape="circle"
         size="middle"
+        disabled={disabled}
         icon={
           <PauseCircleOutlined
             className={'flex justify-center items-center'}
             style={font22px}
           />
         }
-        onClick={() => startInstance(name, idTemplate!)}
+        onClick={() => {
+          if (!disabled) {
+            setDisabled(true);
+            switchPersistent(
+              name,
+              tenantNamespace!,
+              getInstancePatchJson({ running: !running })
+            )
+              .then(() => {
+                setTimeout(() => setDisabled(false), 400);
+              })
+              .catch();
+          }
+        }}
       />
     </Tooltip>
   ) : status === 'VmiOff' ? (
     <Tooltip placement="top" title={'Start'}>
       <Button
+        loading={disabled}
         className={`hidden ${extended ? 'sm:block' : 'xs:block'} py-0`}
         type="success"
         with="link"
         shape="circle"
         size="middle"
+        disabled={disabled}
         icon={
           <PlayCircleOutlined
             className={'flex justify-center items-center'}
             style={font22px}
           />
         }
-        onClick={() => stopInstance(name, idTemplate!)}
+        onClick={() => {
+          if (!disabled) {
+            setDisabled(true);
+            switchPersistent(
+              name,
+              tenantNamespace!,
+              getInstancePatchJson({ running: !running })
+            )
+              .then(() => {
+                setTimeout(() => setDisabled(false), 400);
+              })
+              .catch();
+          }
+        }}
```

Comment:
This code looks replicated. 
Please make a function, also, I'd suggest not making a "toggle function", receive a parameter with the _desired_ running flag and enforce that one. If a button has the pause icon, it has to send `running: false`.
Consider to directly put the button in loading state instead of disabled